src/draw/nema_gfx/lv_draw_nema_gfx_img.c

```diff
@@ -0,0 +1,263 @@
+    }
+}
+
+static void _draw_nema_gfx_tile(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t * dsc, const lv_area_t * coords)
+{
+
+    lv_image_decoder_dsc_t decoder_dsc;
+    lv_result_t res = lv_image_decoder_open(&decoder_dsc, dsc->src, NULL);
+    if(res != LV_RESULT_OK) {
+        LV_LOG_ERROR("Failed to open image");
+        return;
+    }
+
+    int32_t img_w = dsc->header.w;
+    int32_t img_h = dsc->header.h;
+
+    lv_area_t tile_area = *coords;
+    lv_area_set_width(&tile_area, img_w);
+    lv_area_set_height(&tile_area, img_h);
+
+    int32_t tile_x_start = tile_area.x1;
+
+    while(tile_area.y1 <= draw_unit->clip_area->y2) {
+        while(tile_area.x1 <= draw_unit->clip_area->x2) {
+
+            lv_area_t clipped_img_area;
+            if(_lv_area_intersect(&clipped_img_area, &tile_area, draw_unit->clip_area)) {
+                _draw_nema_gfx_img(draw_unit, dsc, &tile_area);
+            }
+
+            tile_area.x1 += img_w;
+            tile_area.x2 += img_w;
+        }
+
+        tile_area.y1 += img_h;
+        tile_area.y2 += img_h;
+        tile_area.x1 = tile_x_start;
+        tile_area.x2 = tile_x_start + img_w - 1;
+    }
+
+    lv_image_decoder_close(&decoder_dsc);
+}
+
+static void _draw_nema_gfx_img(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t * dsc, const lv_area_t * coords)
+{
+    if(dsc->opa <= LV_OPA_MIN) return;
+
+    lv_draw_nema_gfx_unit_t * draw_nema_gfx_unit = (lv_draw_nema_gfx_unit_t *)draw_unit;
+
+    lv_layer_t * layer = draw_unit->target_layer;
+    const lv_image_dsc_t * img_dsc = dsc->src;
+
+    lv_area_t blend_area;
+    /*Let's get the blend area which is the intersection of the area to fill and the clip area.*/
+    if(!_lv_area_intersect(&blend_area, coords, draw_unit->clip_area))
+        return; /*Fully clipped, nothing to do*/
+
+    lv_area_t rel_clip_area;
+    lv_area_copy(&rel_clip_area, draw_unit->clip_area);
+    lv_area_move(&rel_clip_area, -layer->buf_area.x1, -layer->buf_area.y1);
+
+    bool has_transform = (dsc->rotation != 0 || dsc->scale_x != LV_SCALE_NONE || dsc->scale_y != LV_SCALE_NONE);
+    bool recolor = (dsc->recolor_opa > LV_OPA_MIN);
+
+    /*Make the blend area relative to the buffer*/
+    lv_area_move(&blend_area, -layer->buf_area.x1, -layer->buf_area.y1);
+
+    uint32_t tex_w = lv_area_get_width(coords);
+    uint32_t tex_h = lv_area_get_height(coords);
+
+    nema_set_clip(rel_clip_area.x1, rel_clip_area.y1, lv_area_get_width(&rel_clip_area),
+                  lv_area_get_height(&rel_clip_area));
+
+    const void * src_buf = img_dsc->data;
+
+    uint32_t blending_mode = nemagfx_blending_mode(dsc->blend_mode);
+
+    lv_color_format_t src_cf = img_dsc->header.cf;
+
+    //Image contains Alpha
+    if(src_cf == LV_COLOR_FORMAT_ARGB8888 || src_cf == LV_COLOR_FORMAT_XRGB8888) {
+        blending_mode |= NEMA_BLOP_SRC_PREMULT;
+    }
+
+    uint32_t color_format = lv_cf_to_nema(src_cf);
+    uint32_t src_stride = img_dsc->header.stride;
```

Comment:
Remove not used vars.
BTW, does nema support image that has stride?

Reply:
I will remove the unused vars. By stride you mean the number of bytes from one row of pixels in memory to the next row of pixels in memory, right? Can you be more specific ?

Reply:
Yes. The bytes needed to goto next row. The stride can be larger than the real pixel data width.

Reply:
NemaGFX can support multiple image formats, thus multiple strides !